dialogue-core/src/main/java/com/palantir/dialogue/core/BlacklistingChannel.java

```diff
@@ -25,20 +25,32 @@
 import com.palantir.dialogue.Endpoint;
 import com.palantir.dialogue.Request;
 import com.palantir.dialogue.Response;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.UnsafeArg;
 import java.time.Duration;
 import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Delegates execution to the given {@link LimitedChannel} until a failure is seen, at which point the channel will
  * return {@link Optional#empty()} until the given {@link Duration} has elapsed.
+ *
+ * When the blacklisting {@link Duration} has elapsed, the channel enters a 'probation' period, where a few requests
+ * are allowed to be sent off - all of which must be successful for probation to be passed and the channel fully
+ * unblacklisted. Without this functionality, hundreds of requests could be sent to a still-broken
+ * server before the first of them returns and tells us it's still broken.
  */
 final class BlacklistingChannel implements LimitedChannel {
```

Comment:
moving these changes to a dedicated PR: https://github.com/palantir/dialogue/pull/349